lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,29 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "parameter MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
+      merged_options = merge_options(options, other.options)
+
+      IncludeDirective.new(hash, merged_options)
+    end
+
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge!(other)
+      merge_result = merge(other)
```

Comment:
As a general good practice, I tend to favor [defining the "safe" method in terms of the "dangerous" method](https://github.com/bbatsov/ruby-style-guide#safe-because-unsafe) rather than the opposite.